CMakeLists.txt

```diff
@@ -34,11 +34,20 @@ endif()
 
 set(CAPTURE_PACKAGE_VERSION "[ \\t]*\"PACKAGE_VERSION\"[ \\t]*:[ \\t]\"(.*)\"")
 file(STRINGS "properties.json" _faad_version_line REGEX "${CAPTURE_PACKAGE_VERSION}")
-message("${_faad_version_line}")
 string(REGEX REPLACE "${CAPTURE_PACKAGE_VERSION}" "\\1" FAAD_VERSION "${_faad_version_line}")
+string(REPLACE "\." ";" FAAD_VERSION_PARTS ${FAAD_VERSION})
+list(GET FAAD_VERSION_PARTS 0 FAAD_VERSION_MAJOR)
+list(GET FAAD_VERSION_PARTS 1 FAAD_VERSION_MINOR)
+list(GET FAAD_VERSION_PARTS 2 FAAD_VERSION_PATCH)
+# Semantic -> library version
+# NB(eustas): likely that will be always OK; if not, we could read "overrides" from properties.json
+math(EXPR FAAD_ABI_VERSION_CURRENT "${FAAD_VERSION_MAJOR} + ${FAAD_VERSION_MINOR}")
```

Comment:
Does this mean 1.1.0 and 2.0.0 will get the same ABI version?

Reply:
Correct. But they will have different "compatibility".
SO name is `compatibility.age.revision`; if we translate is to semantic values we get exactly `major.minor.patch`
As you can see "current" is dropped, we only calculate it to be honest. And because libtool-version have very logical, but weird mutation rules:
 - interfaces added (or change is compatible)      -> current+1:0:age+1
 - interfaces removed (or changed is incompatible) -> current+1:0:0
 - interfaces not changed                          -> current:revision+1:age

So "current" is somewhat synthetic but useful for finding the range of versions we can use given the version we've been compiled against.